sklearn/neural_network/random_basis_function.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+
+from .base import ACTIVATIONS
+
+from ..base import BaseEstimator, TransformerMixin
+from ..utils import check_random_state
+from ..utils.extmath import safe_sparse_dot
+from ..utils.validation import check_array
+
+
+class RandomBasisFunction(BaseEstimator, TransformerMixin):
+
+    """Random basis activation.
+
+    The algorithm uses the number of features of the input data to randomly 
+    generate coefficient parameters based on the uniform probability 
+    distribution.
+
+    Using these coefficient parameters, the algorithm can transform 
+    the data into a different dimensional space.
+
+    Parameters
+    ----------
+    intercept : boolean, default True
+        Whether to randomly generate an intercept. 
+
+    weight_scale : float, default 'auto'
+        Initializes and scales the input-to-hidden weights.
+        The weight values will range between plus and minus
+        'sqrt(weight_scale * 6. / (n_features + n_hidden))' based on the
+        uniform distribution.
+
+    n_activated_features : int, default 10
```

Comment:
This seems a long name for a common parameter. And it makes it seem like there are non-activated features in the output. What's wrong with `n_outputs`?